### DISCUSSION TITLE
Kanal 0 in allen Geräten?

### DISCUSSION THREAD

**Discussion by jp112sdl** [OpenCCU/OpenCCU#1482] [🇩🇪 Generelle Diskussionen]
Hi Jens,
hattest du das hinzugefügt?
https://github.com/jens-maus/RaspberryMatic/blob/8d8edd35901f338eca23c1035f22f77c33cf849b/buildroot-external/patches/occu/0133-WebUI-Fix-InternalIDExposure.patch#L1-L21

Unter Bedienung->"Räume" bzw. "Gewerke" werden ja nur die :0 Kanäle angezeigt, die man manuell auch dem Raum/Gewerk zugeordnet hat.

Dass jetzt unter "Geräte" bei jedem Gerät der Kanal 0 auftaucht, ist ein bisschen überladen, oder?
![Bildschirmfoto 2021-10-29 um 21 10 15](https://user-images.githubusercontent.com/18190139/139489464-f3e81eba-dbbb-4359-9d0b-2d377bcfa3de.png)

**Comment by jens-maus** [comment]
Ja, das war ich ;) Fand das nur logisch. Und der der die 0:er Kanäle nicht sehen will kann sie ja auf "nicht sichtbar" klickern, oder?

**Comment by Baxxy13** [comment]
Hmm, auch wenn es "Status und Bedienung 'der' Geräte" heißt finde ich die 0er dort fehlplatziert. 
Zumal das ja auch wieder inkonsistent ist da 0er von IP's dort nicht erscheinen. (was idealerweise auch so bleibt!)
Ich habe meistens alle 0er HM/IP in zwei Gewerken (Kanal-0-Battgeräte/Kanal-0-Stromversorgt) primär für die Iteration aber auch mal zum schauen. Vor #1477 tauchten die 0er HM dort ja nicht auf, deshalb das Ticket. 
Der Tipp mit "nicht sichtbar" führt nun dazu das die 0er HM zwar aus "Status und Bedienung > Geräte" verschwinden, aber auch aus den Gewerken/Räumen. Somit habe ich den gleichen Status wie vor #1477.

**Reply by jens-maus** [reply]
Nun, dann sollten wir vielleicht erst einmal darüber reden/diskutieren was jeder hier unter den "Status und Bedienung" Ansichten versteht. In meinem Verständnis sind das lediglich unterschiedliche Sortierungen/Repräsentationen der gleichen Daten. D.h. Also egal ob Geräte, Gewerke oder Räume, es sollten IMHO immer die gleichen Grundbedingungen für die Darstellung angenommen werden und folglich auch die gleichen Kanäle eben anhand der eingestellten "sichtbar"keit in den Einstellungen dargestellt werden.

Konkret bedeutet das, das IMHO eben unter Geräte alle Kanäle (inkl der :0) eines Gerätes dargestellt werden, solange diese als sichtbar markiert wurden unter Einstellungen. Und das gleiche gilt eben auch für Gewerke und Räume, nur das dort eben zusätzlich der Gewerk/Raumfilter noch aktiv ist,d.h also nur die Kanäle im jeweiligen Gewerk/Raum gelistet werden für die dann auch eine Zuordnung in Einstellungen erfolgt ist.

Alles andere stellt für mich nur eine prinzipielle Inkonsistenz dar und erscheint mir unlogisch zu sein. Oder wie erklärt man bitte das die 0: Kanäle unter Geräte nicht auftauchen während sie jedoch unter Gewerk/Raum mit einbezogen werden?!?

**Comment by Baxxy13** [comment]
Kanal:0 ist nun mal ein "Sonderling" der dazu auch noch bei HM/HmIP unterschiedlich behandelt wird.

Im Zuge eines älteren anderen Themas hatte ich damals vorgeschlagen die Routinen für Kanal:0 zwischen HM/HmIP anzugleichen. Das hätte bedeutet Kanal:0 von HM per default auf "nicht intern" zu setzen. Das wurde damals abgelehnt und nun haben wir halt eine inkonsistente Ausgangslage die zu einem konsistenten Ergebnis führen soll.
 

Meine Erklärung wäre das ich mich bei den Räumen/Gewerken explizit dafür entscheide was ich sehen möchte indem ich die entsprechenden Kanäle hinzufüge. Und natürlich das eQ-3 das ursprünglich so designed hat. ;-) 

Letztlich ist es aber wohl eher "Macht der Gewohnheit". Ein HM-Gerät in Status und Bedienung > Geräte anklicken und als erstes den "Servicekanal" zu sehen stört irgendwie den "Flow". Zumal das wie schon geschrieben bei IP nicht der Fall ist. Wenn man den 0er ans untere Ende der Kanäle sortieren könnte würde das vermutlich gar nicht so ins Gewicht fallen.

Habe jetzt mal die oben gezeigten Änderungen rückgängig gemacht und muss sagen das das so nun dem entspricht was ich mir bei dem zugehörigen Issue vorstellte.

**Reply by Baxxy13** [reply]
> Wenn man den 0er ans untere Ende der Kanäle sortieren könnte würde das vermutlich gar nicht so ins Gewicht fallen.

Interessant... jetzt sehe ich gerade das das schon so ist. (Auch auf Jérôme's Screenshot) War wohl eine Browser-Cache Problem bei mir.
![RM_TC_IT_Kanal_0](https://user-images.githubusercontent.com/54898249/139531791-41a61644-982c-41ac-8658-4965a2496d36.JPG)
Ist zwar nicht "schön", aber dann doch ok für mich.

**Comment by jp112sdl** [comment]
"Lass uns mal Kanal 0 mit anzeigen" kam ursprünglich aus der Anmerkung, dass in den Servicemeldungen zu einem Gerät kein Raum/Gewerk steht, was ja damit zusammenhing, dass es nicht möglich war Kanal 0 einem Raum/Gewerk zuzuordnen.

Ob es nun schön ist, dass ein Maintenance-Channel ("Wartung") unter Status und Bedienung auftauchen soll mag Geschmacksache sein.

Für Räume bzw. Gewerke hat das jeder selbst in der Hand, ob er denn einen Kanal 0 hinzufügt oder nicht.

In Geräte ist das jetzt immer da. Und die Maintenance-Informationen machen einen riesen Brocken aus.
Da geht der eigentliche Status eines einfachen Geräts (Schalter EIN/AUS, Zustand OFFEN/ZU) total unter (siehe meinem Screenshot).

**Reply by jens-maus** [reply]
Wie wäre es denn wenn wir bei Status und Bedienung unter Geräte unten nen Button hinzufügen wo man die Maintenance-Kanäle ein/ausblenden lassen kann wie man das auch für virtuelle Kanäle an anderen Stellen dynamisch in einer Ansicht machen kann? Das wäre es IMHO i.O. wenn man die standardmäßig ausblenden lässt und nur bei drücken dieses buttons diese eben einblenden lässt.

**Reply by jp112sdl** [reply]
Unten im Footer?
Ja das wäre doch eine gute Lösung!

**Reply by jens-maus** [reply]
Genau das meine ich, ja.

**Reply by jp112sdl** [reply]
Ok. Du baust den Button ein und ich werde die Tage mal schauen, die Datenpunkte (Kommunikation gestört: Nein: false, [DEVICE_IN_BOOTLOADER],..)   innerhalb des Kanal 0 sinnvoller darzustellen.

**Reply by jens-maus** [reply]
Ich schau mal wann ich dazu komme. Eventl Mitte oder Ende nächster Woche vllt. Wenns schneller sein soll kann gerne jemand anders Hand anlegen ;)

**Comment by Baxxy13** [comment]
Ihr hab ja wieder gute Ideen!
Ich frage lieber gleich vorab...
Wird das jetzt wieder eine HM-Sonderlocke oder ließe sich mit dem Button dann auch der 0er von IP-Geräten ein/ausblenden?

**Reply by jens-maus** [reply]
Das sollten wir anstreben, ja! Da sollte es am besten überall in Status und Bedienung keinerleo Unterschied mehr zwischen Hm und HmIP mehr geben.

**Reply by Baxxy13** [reply]
Ja, ne einheitliche Linie wäre gut.
@jp112sdl ... brauchst du ne VM zum testen wo ein IP-Gerät angelernt ist?
Könnte man ja mal das "Sharing over Tailscale" probieren. Damit sollte kompletter Zugriff über VPN möglich sein und du könntest unabhängig von meiner Anwesenheit dran basteln.

**Reply by jp112sdl** [reply]
@Baxxy13 wo existiert denn eine Kanal-0-Diskrepanz zwischen HM und HmIP?
Nur in Status und Bedienung?

**Reply by Baxxy13** [reply]
Genauer gesagt nur in "Status und Bedienung > Geräte". 
Dort tauchen die 0er von IP trotz `Internal: false` und `Visible:true` (Standard bei IP) nicht auf.
Bei den Gewerken, Räumen und Favoriten sind sie aber zu sehen.

**Reply by jp112sdl** [reply]
Hmm, das kommt von 
https://github.com/eq-3/occu/blob/master/WebUI/www/rega/esp/functions.fn#L387-L391
```
...
    if (chn.HssType().Find("MAINTENANCE") != -1) {
      ! Don't show the maintenance channel of the new HmIP devices
      ! This is necessary because the name of such a channel of an HmIP device
      !   can be MAINTENANCE_BAT_BL, MAINTENANCE_BAT_EL, MAINTENANCE_BAT_CL and so on
      ! For BidCos-RF this isn't necessary
...
```

Ich hab in meiner Produktivumgebung einen HmIP-PCBS , da sieht es dann so aus, wenn den MAINTENANCE mit rein nehme:
<img width="610" alt="Bildschirmfoto 2021-10-31 um 11 29 52" src="https://user-images.githubusercontent.com/18190139/139578360-2b042943-7f8d-49e0-9169-9c9940d7e0fe.png">
Das war es dann aber auch mit dem HmIP Gerätepark bei mir.

Was eQ-3 mit `MAINTENANCE_BAT_BL, MAINTENANCE_BAT_EL, MAINTENANCE_BAT_CL and so on` meint, weiß ich nicht so recht.
Die `if`-Abfrage hat dann auch Folgeroutinen für `HmIP-W`.

Kanal 0 steht bei HmIP dann auch oben als erster Kanal und bei HM, CUxD etc. als letzter.
Da weiß ich grad noch nicht woher die Sortierung kommt.

**Reply by Baxxy13** [reply]
Hmm, wälzt man die IP-Device-Doku tauchen die erwähnten HSSTypen bei verschiedensten Geräten auf. Bspw. beim STHD. Interessanterweise zeigt mir der SDV bei meinen 2 STHD's (entgegen der Doku) das übliche "MAINTENANCE" bei ChnLabel und HssType.
Ich müsste meinen Gerätepark mal durchiterieren um zu schauen ob da was auftaucht.

Das mit der Sortierung findet sich bestimmt auch noch.

**Reply by Baxxy13** [reply]
Die Iteration über alle meine IP-Geräte hat keine HssTypen ala...
`MAINTENANCE_BAT_BL, MAINTENANCE_BAT_EL, MAINTENANCE_BAT_CL`
aufgedeckt. Das heißt natürlich erstmal nicht viel.
Könnten wir ja vorerst ignorieren bis jemand fragt warum von IP-Gerät XYZ der 0er nicht auftaucht. ;-)

**Reply by jp112sdl** [reply]
Hab die Änderung mal in den PR #1488 mit reingepackt.

**Reply by Baxxy13** [reply]
Fehlt da nicht ne `{` am Ende der if-Abfrage?
Edit:
Ah ok, geht auch ohne {. 
IP 0er werden angezeigt, fehlt nur noch die Sortierung ans Ende. 
Soweit so gut.

**Reply by jp112sdl** [reply]
Jup, hab ich geändert

**Comment by jp112sdl** [comment]
Ich habe jetzt erstmal die Ansicht des Kanal 0 verbessert. #1487
Weitere HTML-Anpassungen überlasse ich mal Jens 😛 

Und auch den Footer-Button habe ich implementiert, analog zum "interne Systemvariablen einblenden" Button.
Der Linter meckert jedoch, dass der Patch 0133 nicht die `devices.htm` patchen kann. Hmm... #1488

**Comment by Baxxy13** [comment]
Ich schreibe mal hier um den PR übersichtlich zu halten.

Edit:
Ausführungen in den PR verschoben

**Reply by jens-maus** [reply]
> Ich schreibe mal hier um den PR übersichtlich zu halten.

Der muss nicht "übersichtlich" sein. Es stört aber IMHO mehr wenn es jetzt neben Homematic Forum und PR noch ne dritte stelle gibt wo diskutiert wird :) IMHO Gehört diese gesamte diskussion in den PR rein und nicht in den diskussionsteil hier. Deshalb gibt es in dem GitHub Discussionsteil ja auch keine explizite Entwicklungsdiskussionsordner.

**Reply by Baxxy13** [reply]
Alles klar, ich "verschiebe" es.

**Reply by jp112sdl** [reply]
Nun ja, den Kanal 0 als solches überhaupt irgendwo anzuzeigen (sei es händisch zugeordnet in einem Raum/Gewerk oder grundsätzlich in Geräte), halte ich schon die ganze Zeit für wenig sinnvoll.

Welchen Nutzen hat der Kanal im täglichen Gebrauch? 
CONFIG_PENDING, UPDATE_PENDING, DEVICE_IN_BOOTLOADER, LOWBAT, UNREACH, STICKY_UNREACH bekommt man alles als Servicemeldung, wenn etwas vorliegt.
Bleiben nur noch die RSSI-Werte und die sieht man auch in Einstellungen->Geräte

Zumindest bin ich froh, dass der Kanal nur ein Opt-In über den Footer-Button wird.

**Reply by Baxxy13** [reply]
> Welchen Nutzen hat der Kanal im täglichen Gebrauch?

Hmm, gute Frage. Einen konkreten dauerhaften Nutzen gibt es wohl nicht. Aber das trifft ja auch auf alle IP Eingangs/Tasterkanäle zu. Diese "haben hier keine Funktion" und werden trotzdem angezeigt. ;-)
Ich nutze den 0er bei IP-Batt-Geräten hauptsächlich für eine schnelle Übersicht der Batt-Spannungen (Gewerk "Batterieversorgt"). Gewerk öffnen, drüberschauen und gut is.
